libs/core/algorithms/include/hpx/parallel/container_algorithms/partition.hpp

```diff
@@ -134,56 +127,59 @@ namespace hpx { namespace ranges {
     /// within each thread.
     ///
     /// \returns  The \a partition algorithm returns a \a
-    ///           hpx::future<subrange_t<FwdIter>>
+    ///           hpx::future<subrange_t<hpx::traits::range_iterator_t<Rng>>>
     ///           if the execution policy is of type \a parallel_task_policy
-    ///           and returns \a subrange_t<FwdIter> otherwise.
+    ///           and returns \a subrange_t<hpx::traits::range_iterator_t<Rng>>
     ///           The \a partition algorithm returns a subrange starting with
     ///           an iterator to the first element of the second group and
     ///           finishing with an iterator equal to last.
     ///
-    template <typename ExPolicy, typename FwdIter, typename Sent,
-        typename Pred, typename Proj>
-    typename util::detail::algorithm_result<ExPolicy,
-        subrange_t<FwdIter>>::type
-    partition(ExPolicy&& policy, FwdIter first, Sent last, Pred&& pred,
-        Proj&& proj);
+    template <typename ExPolicy, typename Rng, typename Pred,
+        typename Proj = parallel::util::projection_identity>
+    typename parallel::util::detail::algorithm_result<ExPolicy,
+        subrange_t<hpx::traits::range_iterator_t<Rng>>>::type
+    partition(ExPolicy&& policy, Rng&& rng, Pred&& pred, Proj&& proj = Proj());
```

Comment:
Usually in the function declaration we have `Pred&& pred = Pred()` and not just `Pred&& pred`. Why is that here?

Reply:
Yes, ranges::partition requires a predicate (see https://eel.is/c++draft/alg.partitions#lib:is_partitioned)